test/distributed/test_jit_c10d.py

```diff
@@ -66,3 +80,30 @@ def run_pg_nccl_alltoall(
             return work.result()
 
         run_pg_nccl_alltoall(nccl_pg, output, input)
+
+    @requires_nccl()
+    @skip_if_rocm_single_process
+    def test_init_process_group_nccl_as_base_process_group_torchbind(self):
+        self._create_nccl_pg_as_base_process_group()
+
+    @requires_nccl()
+    @skip_if_rocm_single_process
+    def test_process_group_nccl_as_base_process_group_torchbind_alltoall(self):
+        nccl_pg = self._create_nccl_pg_as_base_process_group()
+
+        input = torch.rand(16).cuda()
+        output = torch.rand(16).cuda()
+
+        @torch.jit.script
+        def run_pg_nccl_alltoall(
+            pg: torch.classes.dist_c10d.ProcessGroup,
+            output: torch.Tensor,
+            input: torch.Tensor
+        ):
+            output_split_sizes: List[int] = []
+            input_split_sizes: List[int] = []
+            work = pg.alltoall_base(output, input, output_split_sizes, input_split_sizes)
+            work.wait()
+            return work.result()
+
+        run_pg_nccl_alltoall(nccl_pg, output, input)
```

Comment:
can you also test ProcessGroupGloo as well? given the `newProcessGroupHelper` also creates gloo pg.

Reply:
It would require TorchBinding ProcessGroupGloo's Options, which we don't have and don't need currently.